pkg/helpers/helpers.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package helpers
+
+import (
+	"fmt"
+	"math/big"
+	"net"
+)
+
+// IsIPInRange checks if the target IP falls within the start and end IP range (inclusive).
+func IsIPInRange(startIP, endIP, targetIP string) (bool, error) {
+	// Parse the IPs
+	start := net.ParseIP(startIP)
+	end := net.ParseIP(endIP)
+	target := net.ParseIP(targetIP)
+
+	// Ensure all IPs are valid
+	if start == nil {
+		return false, fmt.Errorf("invalid start IP: %q", startIP)
+	}
+	if end == nil {
+		return false, fmt.Errorf("invalid end IP: %q", endIP)
+	}
+	if target == nil {
+		return false, fmt.Errorf("invalid target IP: %q", targetIP)
+	}
+
+	// Convert IPs to big integers
+	startInt := ipToBigInt(start)
+	endInt := ipToBigInt(end)
+	targetInt := ipToBigInt(target)
+
+	// Check if target IP is within the range
+	return targetInt.Cmp(startInt) >= 0 && targetInt.Cmp(endInt) <= 0, nil
+}
+
+// ipToBigInt converts a net.IP to a big.Int for comparison.
+func ipToBigInt(ip net.IP) *big.Int {
+	// Normalize to 16-byte representation for both IPv4 and IPv6
+	ip = ip.To16()
+	return big.NewInt(0).SetBytes(ip)
+}
```

Comment:
Let's use [`net/netip`](https://pkg.go.dev/net/netip) and [`go4.org/netipx`](https://pkg.go.dev/go4.org/netipx) instead:
```suggestion
import (
	"fmt"
	"net/netip"
	
	"go4.org/netipx"
)

// IsIPInRange checks if the target IP falls within the start and end IP range (inclusive).
func IsIPInRange(startIP, endIP, targetIP string) (bool, error) {
	// Parse the IPs
	start, err := netip.ParseAddr(startIP)
	if err != nil {
		return false, fmt.Errorf("invalid start IP %q: %w", start, err)
	}
	end, err := netip.ParseAddr(endIP)
	if err != nil {
		return false, fmt.Errorf("invalid end IP %q: %w", end, err)
	}
	target, err := netip.ParseAddr(targetIP)
	if err != nil {
		return false, fmt.Errorf("invalid target IP %q: %w", target, err)
	}
	
	ipRange := netips.IPRangeFrom(start, end)

	// Check if target IP is within the range
	return ipRange.Contains(target), nil
}
```

Reply:
Thanks. I vaguely knew there was a `.Compare`, `.Less` methods, but I couldn't find it, so thought I might be mistaken.
Nit: Did not feel importing ext. pkg `"go4.org/netipx"` was adding any value so used only `"net/netip"`.